tx-query/app/src/enclave_u/mod.rs

```diff
@@ -1,19 +1,229 @@
+use chain_core::tx::data::TxId;
+use enclave_u_common::{storage_path, TX_KEYSPACE};
+use log::{debug, error, trace};
+use parity_scale_codec::{Decode, Encode};
 use sgx_types::*;
+use sled::{ConfigBuilder, Db, Tree};
+use std::net::{SocketAddr, TcpStream};
+use std::os::unix::io::IntoRawFd;
+use std::sync::Arc;
+
+fn init_tx_db() -> Arc<Tree> {
+    let config = ConfigBuilder::default()
+        .path(storage_path())
+        .read_only(true);
+
+    let db = Db::start(config.build()).expect("failed to open a storage path");
+    db.open_tree(TX_KEYSPACE)
+        .expect("failed to open a tx keyspace")
+}
+
+thread_local! {
+    static TXDB: Arc<Tree> = init_tx_db();
```

Comment:
Why to wrap `Tree` in an `Arc` if it is `thread_local`?

Reply:
That's what's returned from `db.open_tree(TX_KEYSPACE)`: https://docs.rs/sled/0.25.0/sled/struct.Db.html#method.open_tree

Reply:
But I guess it may go away https://github.com/crypto-com/chain-tx-enclave/pull/24#discussion_r315122508